src/uaa/uaa.js

```diff
@@ -26,6 +27,32 @@ export default class UAAClient {
     return this.accessToken;
   }
 
+  async getSigningKey() {
+    if (this.signingKey) {
+      return this.signingKey;
+    }
+
+    const response = await axios.request({
+      url: '/token_keys',
+      method: 'get',
+      baseURL: this.apiEndpoint,
+      validateStatus: status => status > 0 && status < 501
```

Comment:
Why `< 501` here?

Reply:
The library that makes the requests (`axios`), handles everything for us. We found the thrown errors not particularly helpful, therefore we'd like it (the library) to "successfully" finish all requests and leave it up to us, to deal with them later.

Following if statement attempts to build a better message for easier recognition.

This may change overtime as we'd slowly get out of the heavy dev mode.

Reply:
We spoke about this In Real Life. The thing I found confusing here was that a `500` error was treated differently than a `501`. The ultimate behaviour would have been the same because of the next stanza but I felt it read a bit weird.

Reply:
I'll drop it to `500`. It's "unfair" for the other `5xx` :D